filename_formatter.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+echo -e "Arguments:
+[0] - Used by Bash (script filename)
+[1] - Base directory
+[2] - Filename type (maximum two values)
+[3] - Ignored files or folders (optional; use "\""./<directory_name>"\"")
+"
+
+echo "Changed files:"
+
+# Separate $2 value (filename types) if it has a comma
+if [[ "$2" == *","* ]];
+then
+string="$2"
+
+str_value=${string#*,}
+str_value2=${string%%,*}
+else
+str_value="$2"
+str_value2="$2"
+fi
+
+IFS=$'\n'
+for fname in `find $1 -type f -name "$str_value2" -o -name "$str_value"`
+do
+    ignored_files="$(echo $3 | tr "," "\n")"
+
+    str="${fname}"
+    value=${str%/*} # If the base directory is `.`, check in all directories for the ignored filenames
+
+    for files in $ignored_files
+    do
+        if [ "${fname}" == "$value/$files" ] || [ "$value" == "$files" ];
+        then
+            continue 2
+        fi
+    done
+
+    echo ${fname}
+    new_fname=`echo ${fname} | tr ' ' '_'`
+    echo "      ${new_fname}"
+    new_fname=`echo ${new_fname} | tr 'A-Z' 'a-z'`
+    echo "      ${new_fname}"
+    new_fname=`echo ${new_fname} | tr '-' '_'`
+    echo "      ${new_fname}"
```

Comment:
Do we need these `echo`s?

Reply:
Not really, it is to show the different ways how you could format it, I guess. We can remove them if needed.

Reply:
I think it's more like a pipeline - first it replaces all "space" characters to "underscores" then it replaces alphabets name to lower case and finally it replaces the "hyphens" to "underscores". All this is not really useful to the user I think, let me know if you agree/disagree 😄. Maybe if we want, we can have a verbose flag `-v` or something to make the output very verbose and have theses echos when it's turned on.

Reply:
Yeah. I'll be marking the `echo`'s as comments for now. Thanks. :)